lib/spack/spack/cmd/ci.py

```diff
@@ -341,12 +341,12 @@ def ci_rebuild(args):
     # also be used in the generated "spack install" command to install the spec
     tty.debug('job concrete spec path: {0}'.format(job_spec_yaml_path))
     with open(job_spec_yaml_path, 'w') as fd:
-        fd.write(job_spec.to_yaml(hash=ht.build_hash))
+        fd.write(job_spec.to_yaml(hash=ht.full_hash))
```

Comment:
Make this write JSON now that #22845 is in.

Reply:
Yes, will do.